### ISSUE TITLE
Repo: Typescript 5.1.3 support

### ISSUE THREAD

**Issue by RobertGemmaJr** [typescript-eslint/typescript-eslint#7080]
### Suggestion

Getting a warning when using the most up to date version of typescript (5.1.3). Bug fix here so it should be a quick update (:

```shell
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.1.0

YOUR TYPESCRIPT VERSION: 5.1.3

Please only submit bug reports when using the officially supported version.

=============
```

**Comment by JoshuaKGoldberg** [comment]
This is a duplicate of #6934. Before filing an issue, please [use our issue search](https://github.com/typescript-eslint/typescript-eslint/issues) to check for open and closed issues that already address what you're looking for.
